File: Python3/647. Palindromic Substrings.py
class Solution:
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        length = len(s) # 136ms
        ans = 0
        for center in range(2*length-1):
            left = center // 2
            right = left + center%2
            while left >= 0 and right < length and s[left] == s[right]:
                ans += 1
                left -= 1
                right += 1
        return ans

File: Golang/647. Palindromic Substrings.go
func countSubstrings(s string) int {
    length := len(s)
    ans := 0
    for center := 0; center < 2*length-1; center++ {
        left := center / 2
        right := left + center%2
        for left >= 0 && right < length && s[left] == s[right] {
            ans++
            left--
            right++
        }
    }
    return ans
}
